src/planner/Maintain.h

```diff
@@ -341,71 +341,180 @@ class DropSchema : public SingleInputNode {
+                             std::string name,
+                             bool ifExists) {
+        return qctx->objPool()->add(new DropTagNode(qctx->genId(), input,
+                                                    std::move(name), ifExists));
     }
 
 private:
-    DropTag(int64_t id,
-            PlanNode* input,
-            std::string name,
-            bool ifExists)
-        : DropSchema(id, Kind::kDropTag, input, std::move(name), ifExists) {
+    DropTagNode(int64_t id,
+                PlanNode* input,
+                std::string name,
+                bool ifExists)
+        : DropSchemaNode(id, Kind::kDropTag, input, std::move(name), ifExists) {
     }
 };
 
-class DropEdge final : public DropSchema {
+class DropEdgeNode final : public DropSchemaNode {
 public:
-    static DropEdge* make(QueryContext* qctx,
-                          PlanNode* input,
-                          std::string name,
-                          bool ifExists) {
-        return qctx->objPool()->add(new DropEdge(qctx->genId(), input, std::move(name), ifExists));
+    static DropEdgeNode* make(QueryContext* qctx,
+                              PlanNode* input,
+                              std::string name,
+                              bool ifExists) {
+        return qctx->objPool()->add(new DropEdgeNode(qctx->genId(), input,
+                                                     std::move(name), ifExists));
     }
 
 private:
-    DropEdge(int64_t id,
-             PlanNode* input,
-             std::string name,
-             bool ifExists)
-        : DropSchema(id, Kind::kDropEdge, input, std::move(name), ifExists) {
+    DropEdgeNode(int64_t id,
+                 PlanNode* input,
+                 std::string name,
+                 bool ifExists)
+        : DropSchemaNode(id, Kind::kDropEdge, input, std::move(name), ifExists) {
     }
 };
 
-class CreateTagIndex final : public SingleInputNode {
+// class CreateIndexNode : public SingleInputNode {
+// protected:
+//     CreateIndexNode(int64_t id,
+//                     PlanNode* input,
+//                     Kind kind,
+//                     std::string name,
+//                     meta::cpp2::Schema schema,
+//                     bool ifNotExists)
+//         : SingleInputNode(id, kind, input)
+//         , name_(std::move(name))
+//         , schema_(std::move(schema))
+//         , ifNotExists_(ifNotExists) {}
+
+// public:
+//     const std::string& getName() const {
+//         return name_;
+//     }
+
+//     const meta::cpp2::Schema& getSchema() const {
+//         return schema_;
+//     }
+
+//     bool getIfNotExists() const {
+//         return ifNotExists_;
+//     }
+
+//     std::unique_ptr<cpp2::PlanNodeDescription> explain() const override;
+
+// protected:
+//     std::string            name_;
+//     meta::cpp2::Schema     schema_;
+//     bool                   ifNotExists_;
+// };
+
+class CreateTagIndexNode final : public SingleInputNode {
```

Comment:
Why remove the `CreateIndex` class? I think it's more natural to reuse some code that way.

Reply:
Yep Actually this one is still processing and I push it out just wish to avoid some one implement it duplicated.